### PR TITLE
:recycle: Fix deepsource: VET-V0008 lock erroneously passed by value (internal/net)

### PR DESCRIPTION
**Pull request by vankichi**
### Description:

I have fixed `VET-V0008` under the `internal/net` package.

### Versions:

- Go Version: 1.19.2
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 1.14.8

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document and completed [our CLA agreement](https://cla-assistant.io/vdaas/vald).
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?

### CODE REVIEW

File: internal/net/grpc/client_test.go [hunk 1]
```diff
@@ -114,11 +126,15 @@ func Test_gRPCClient_StartConnectionMonitor(t *testing.T) {
...
-		conns               grpcConns
+		addrs       map[string]struct{}
+		atomicAddrs AtomicAddrs
+		poolSize    uint64
+		clientCount uint64
+		conns       struct {
+			read   atomic.Value
+			dirty  map[string]*entryGrpcConns
+			misses int
+		}
```
Comment (vankichi) [inline internal/net/grpc/client_test.go:137]: I defined the struct which does NOT contain `mu`

File: internal/net/grpc/addrs_test.go [hunk 2]
```diff
@@ -19,15 +19,14 @@ package grpc
 
 import (
 	"reflect"
-	"sync"
 	"sync/atomic"
 	"testing"
 
-	"github.com/vdaas/vald/internal/errors"
+	"github.com/pkg/errors"
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/net/grpc/addrs_test.go:25]: is it correct?
```suggestion
	"github.com/vdaas/vald/internal/errors"
```

[the remark this answers is not in the document]
Comment (vankichi) [reply internal/net/grpc/addrs_test.go]: Fixed

File: internal/net/dialer_test.go [hunk 3]
```diff
@@ -266,14 +265,20 @@ func TestNewDialer(t *testing.T) {
...
 			cmp.AllowUnexported(*want),
 			cmp.Comparer(func(x, y cacher.Cache) bool {
 				if x == nil && y == nil {
 					return true
 				}
-				return !(x == nil && y != nil) || !(y == nil && x != nil)
+				if x == nil || y == nil {
+					return false
+				}
+				return reflect.DeepEqual(x, y)
```
Comment (kevindiu) [inline internal/net/dialer_test.go:281]:
```suggestion
				return reflect.DeepEqual(x, y)
```

**Review by kevindiu — APPROVED** [review]

**Review by datelier — APPROVED** [review]

**Pull request merged**